src/Npgsql/KerberosUsernameProvider.cs

```diff
@@ -52,15 +52,15 @@ static void DetectUsername()
                 return;
             }
 
-            var line = "";
+            var line = default(string);
             for (var i = 0; i < 2; i++)
                 if ((line = process.StandardOutput.ReadLine()) == null)
                 {
                     Log.Debug("Unexpected output from klist, aborting Kerberos username detection");
                     return;
                 }
 
-            var components = line.Split(':');
+            var components = line!.Split(':');
```

Comment:
I think it's better to do a runtime check for null and exit cleanly with a Log.Debug like other cases in this code.

Reply:
If `line` is `null` it already happens in lines 57-61. No need to check it twice.